### ISSUE TITLE
Chummer4 + android

### ISSUE THREAD

**Issue by loikki** [chummer5a/chummer5a#452]
Hi,
I d like to do some modifications on chummer4 but all the forks are more or less dead (except the one of Fizztastic but he is not doing anything else than accepting a pull request), therefore I was thinking about working on this one (I can trying to merge the code from Fizztastic with your before working on it).

I was thinking about creating an android app (it will avoid people from taking a computer everywhere), but it would be too much work to do it by myself. Therefore are you interested by porting this code to android with my help?

**Comment by joha4270** [comment]
First of all, this is the issue tracker for Chummer5a, you should look over [here](github.com/chummer5a/chummer) for chummer4.

Second, creating an android app based on chummer would be a lot of work, more that we would like to commit to, for multiple reasons.

First of all, the UI code and the logic is not properly separated, meaning that have to be done before porting can be started. We are talking about refactoring some 40k ish lines and a architectural change to really do it properly. If this was Chummer5a i wouldn't mind doing some of it, but Chummer4 is all but dead and only exists for historical reasons.

Secondly, chummer is written in C#, and therefore not running natively on android. There is tools to work around that ([Xamarin](https://xamarin.com/starter)) but is bigger than what their free licence allows (well, i'm 99% sure the raw code is bigger than 128k compiled) and paying 25 usd/month is atleast not an option for me, being on a student budget.

That being said, creating either a reader that understands .chum(5?) files, or a full blown port, go ahead. If you want to know how specific things work, you can ask, but it is limited how many resources we will devote to that task

**Comment by loikki** [comment]
Sorry for the wrong issue tracker.
Thank you for the answer. 
I did not look close enough the code, I was expecting something more easily port to java.
I will think about it, but without help, I won't do an android apps.
